iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/thread.hpp

```diff
@@ -28,10 +33,58 @@ constexpr uint64_t MAX_THREAD_NAME_LENGTH = 15U;
 
 using ThreadName_t = cxx::string<MAX_THREAD_NAME_LENGTH>;
 
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept;
-ThreadName_t getThreadName(pthread_t thread) noexcept;
+enum class ThreadError
+{
+    EMPTY_CALLABLE,
+    INSUFFICIENT_MEMORY,
+    INSUFFICIENT_PERMISSIONS,
+    INSUFFICIENT_RESOURCES,
+    INVALID_ATTRIBUTES,
+    UNDEFINED
+};
+
+class Thread
+{
+  public:
+    using callable_t = cxx::function<void()>;
+
+    Thread(const Thread&) = delete;
+    Thread& operator=(const Thread&) = delete;
+    Thread(Thread&& other) = delete;
```

Comment:
This goes also for `semaphores` and `mutex` and here the solution would be to acquire some memory outside of the class and store the object in there.
One could change this but then we have to solve a race condition at the start which makes this object much more complex and the benefit is litte. For us it would be actually non existing since we do not have this use case in iceoryx.

Here the problem:
```
posix::Thread t(someFunction);
posix::Thread t2 = std::move(t); // what if the thread is started but the function argument is not yet copied or called inside of `startRoutine`.
```
We could of course make the ctor blocking until the callable is copied and started inside of `startRoutine`. But how? We could add a semaphore (which can fail an then we have to add real complex error handling). A busy loop which burns the CPU?

Or we leave it as it is for now and see if we really need this feature.